src/USER-BROWNIAN/fix_brownian.cpp

```diff
@@ -0,0 +1,217 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
```

Comment:
Here is another idea to ponder, but not a requirement: since there seems to be a significant amount of shared code, it might be conceivable to have all three brownian dynamics integrators implemented in a shared base class and then the three variants just thin wrappers. See fix_nh.cpp vs. fix_nvt.cpp and fix_npt.cpp for example. Please also see my comment about avoiding performance critical if statements in inner loops through template functions. The choice point vs. sphere vs. ellipsoid could be a second "dimension", i.e. a second template option.

Reply:
Sort of did this. The code became a bit more complex when I corrected my treatment of anisotropic rotational diffusion, so I don't think it separated as nicely after all.